besu/src/main/java/org/hyperledger/besu/services/BesuPluginContextImpl.java

```diff
@@ -69,6 +70,10 @@
     serviceRegistry.put(serviceType, service);
   }
 
+  public void addServicesFrom(final PluginServiceFactory factory) {
+    factory.appendPluginServices(this);
```

Comment:
nit: Why we we have this level of indirection?  Wouldn't it be easier to remove this method and just call `factory.appendPluginServices(besuPluginContext);` from the calling site?

Reply:
that is true in retrospect